kratos/processes/calculate_nodal_area_process.cpp

```diff
@@ -0,0 +1,160 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//  Collaborators:   Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "utilities/geometry_utilities.h"
+#include "processes/calculate_nodal_area_process.h"
+
+namespace Kratos
+{
+
+template<bool THistorical>
+void CalculateNodalAreaProcess<THistorical>::Execute()
+{
+    KRATOS_TRY
+
+    // Check if variables are available
+    KRATOS_ERROR_IF(mrModelPart.Nodes().size() == 0) << "No nodes in the model part" << std::endl;
+    if (THistorical) {
+        KRATOS_ERROR_IF_NOT(mrModelPart.NodesBegin()->SolutionStepsDataHas( NODAL_AREA )) << "Variable NODAL_AREA not in the model part!" << std::endl;
+    }
+
+    // Set to zero the nodal area
+    const auto it_node_begin = mrModelPart.NodesBegin();
+    #pragma omp parallel for
+    for(int i=0; i<static_cast<int>(mrModelPart.Nodes().size()); ++i) {
+        auto it_node = it_node_begin + i;
+        SetInitialValue(it_node);
```

Comment:
```suggestion
        SetAreaValue(*(*it_node), 0.0); // not entirely sure, you might need "get"
```

this way you don't need the version of the function with the iterator

btw isn't this in VariableUtils?

Reply:
Can we live without this?, it is a very small change, but it implies to pass again all the checks, and lately are failing for time limitations a lot

Reply:
actually this is in the variableutils:
~~~
    if (THistorical) {
        VariableUtils().SetVariable(NODAL_AREA, 0.0, mrModelPart.Nodes());
    } else {
        VariableUtils().SetNonHistoricalVariable(NODAL_AREA, 0.0, mrModelPart.Nodes());
}
~~~

=> then you can remove the iterator-function completely